studio/src/components/graph-visualization.tsx

```diff
@@ -107,6 +116,35 @@ function GraphVisualization({
   const [nodes, setNodes] = useState<Node[]>([]);
   const [edges, setEdges] = useState<Edge[]>([]);
 
+  const [showAll, setShowAll] = useState(false);
+  const [topCategory, setTopCategory] = useState("latency");
+
+  const subgraphs = useMemo(() => {
+    if (showAll) {
+      return graphData?.subgraphs ?? [];
+    }
+
+    return (
+      graphData?.subgraphs.slice(0, 5).sort((a, b) => {
```

Comment:
Why do we need to calculate metrics in the frontend?

Reply:
Got it. You only sort it.

Reply:
We already have all the data. Local sort is fast. Server side sort will be overkill for this.